src/librustc_mir/const_eval.rs

```diff
@@ -76,7 +75,10 @@ pub fn op_to_const<'tcx>(
             _ => false,
         };
     let normalized_op = if normalize {
-        ecx.try_read_immediate(op)?
+        match ecx.try_read_immediate(op)? {
```

Comment:
You could use `ecx.try_read_immediate(op)?.deref()`: https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.deref.